### PR TITLE
#412

### PR DESCRIPTION
**Pull request by HDouss**
Resolving issue #412 
The approach is to base on the xsl in order to have the metric output structure, and then to update relevant values using Java calc. These values are especially number of pairs (`class/vars/var[@id=pairs]/text()`) and the actual LCOM4 metric (`class@value`)

### CODE REVIEW

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) everybody who has role `REV` is banned at [#414](https://github.com/yegor256/jpeek/issues/414); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by HDouss** [comment]
@paulodamaso ping

File: src/main/java/org/jpeek/Lcom4Calculus.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek;
```
Comment (paulodamaso) [inline src/main/java/org/jpeek/Lcom4Calculus.java]: @HDouss Let's move all java classes into a new package, `org.jpeek.calculus`, and put all calculus implemented in java in `org.jpeek.calculus.java`  and the XSL ones in `org.jpeek.calculus.xsl`

File: src/main/java/org/jpeek/Lcom4Calculus.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.Map;
+import org.cactoos.io.ResourceOf;
+import org.cactoos.text.Joined;
+import org.cactoos.text.TextOf;
+
+/**
+ * LCOM4 Metrics java calculus.
+ * @since 0.30.9
+ */
+public final class Lcom4Calculus implements Calculus {
```
Comment (paulodamaso) [inline src/main/java/org/jpeek/Lcom4Calculus.java]: @HDouss If we name the package like above, we can change the name of this class to `LCOM4`

File: src/test/java/org/jpeek/Lcom4CalculusTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.jpeek.skeleton.Skeleton;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+
+/**
+ * Test case for {@link Lcom4Calculus}.
+ * @since 0.30.9
+ */
+public final class Lcom4CalculusTest {
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/Lcom4CalculusTest.java]: @HDouss We are not testing if the metric is being correctly calculated. Please add tests or a puzzle so we can assure that we are calculating the value correctly.

File: src/test/java/org/jpeek/Lcom4CalculusTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek;
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/Lcom4CalculusTest.java]: @HDouss Like said above, please change the package of the tests according to the production classes

File: src/test/java/org/jpeek/Lcom4CalculusTest.java (hunk 3 above)
Comment (HDouss) [reply src/test/java/org/jpeek/Lcom4CalculusTest.java]: @paulodamaso the test `methodMethodCalls` is testing this. May be we should add more cases.

**Comment by HDouss** [comment]
@paulodamaso Thanks! Fixed.

**Comment by HDouss** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@HDouss Thank you!

**Comment by paulodamaso** [comment]
@rultor release tag is `0.30.9`

**Comment by 0crat** [comment]
Job `gh:yegor256/jpeek#414` is not assigned, can't get performer

**Pull request merged**